### PR TITLE
ggally_sumarise_by(), ggally_crosstable() and ggtable()

### PR DESCRIPTION
**Pull request by larmarange**
The purpose of `ggtable()` is to reproduce the need raised in #317.

For this, I developed a variation of `ggally_table()` called `ggally_table2()`. In addition, I'm also proposing in this PR the function `ggally_summarised_by()` previously mentionned.

Of course, a dedicated vignette for `ggtable()` will be proposed soon.

### CODE REVIEW

File: R/stat_cross.R [hunk 1]
```diff
@@ -301,3 +301,72 @@ ggally_tableDiag <- function(data, mapping, ..., geom_tile_args = NULL) {
...
+#'
+#' # fill cells with standardized residuals
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), fill = "stdres")
+#'
+#' \dontrun{
+#' # change scale for fill
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), fill = "stdres") +
+#'   scale_fill_steps2(breaks = c(-2, 0, 2), show.limits = TRUE)
+#' }
+ggally_table2 <- function(
```
Comment (schloerke) [inline R/stat_cross.R:339]: May we call this `ggally_crosstable` (or `ggally_cross`)? ... Something without the `2`.  When I see the `2`, I interpret that I should not use the original.

File: R/stat_cross.R [hunk 2]
```diff
@@ -301,3 +301,72 @@ ggally_tableDiag <- function(data, mapping, ..., geom_tile_args = NULL) {
...
+#'
+#' \code{ggally_table2} is a variation of \code{ggally_table} with few modifications: (i) table cells are drawn; (ii) x and y axis are not expanded (and therefore are not aligned with other \code{ggally_*} plots); (iii) content and fill of cells can be easily controlled with dedicated arguments.
+#' @param data data set using
+#' @param mapping aesthetics being used
+#' @param cells Which statistic should be displayed in table cells?
+#' @param fill Which statistic should be used for filling table cells?
+#' @param ... other arguments passed to \code{\link[ggplot2]{geom_text}(...)}
+#' @param geom_tile_args other arguments passed to \code{\link[ggplot2]{geom_tile}(...)}
+#' @importFrom scales percent
+#' @importFrom scales number
```
Comment (schloerke) [inline R/stat_cross.R:315]:
```suggestion
```

Not needed as they are namespaced.

File: R/gg-plots.R [hunk 3]
```diff
@@ -1688,3 +1688,131 @@ ggally_autopointDiag <- function(data, mapping, ...) {
...
+#'
+#' # custom stat function
+#' weighted_sum <- function(x, weights = NULL) {
+#'   if (is.null(weights)) weights <- 1
+#'   paste0("Total : ", round(sum(x * weights, na.rm = TRUE), digits = 1))
+#' }
+#' ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day), text_fn = weighted_sum)
+#'
+#'
+#' @importFrom scales label_percent
```
Comment (schloerke) [inline R/gg-plots.R:1729]:
```suggestion
```

Not needed as it is namespaced

File: R/gg-plots.R [hunk 4]
```diff
@@ -1688,3 +1688,131 @@ ggally_autopointDiag <- function(data, mapping, ...) {
...
+    weight <- eval_data_col(data, mapping$weight)
+    l <- c()
+    for (i in unique(y)) {
+      l <- c(l, text_fn(x[y == i], weight[y == i]))
+    }
+    res <- data.frame(
+      y = unique(y),
+      label = l,
+      stringsAsFactors = FALSE
+    )
```
Comment (schloerke) [inline R/gg-plots.R:1753]:
```suggestion
    res <- ddply(
      data.frame(
        x = eval_data_col(data, mapping$x),
        y = eval_data_col(data, mapping$y),
        weight = eval_data_col(data, mapping$weight) %||% 1,
        stringsAsFactors = FALSE
      ),
      c('y'),
      plyr::here(summarize),
      label = text_fn(x, weight)
    )
```

File: R/stat_cross.R (hunk 1 above)
Comment (larmarange) [reply R/stat_cross.R:339]: names changed for ggally_crosstable

File: R/stat_cross.R (hunk 2 above)
Comment (larmarange) [reply R/stat_cross.R:315]: fixed

**Comment by larmarange** [comment]
Requested changes have been made (including name change).

stat_cross() has been modified for proper handling of cells with zero observations.

a vignette for `ggtable()` has been added

**Pull request merged**